src/main/java/spireMapOverhaul/zones/gremlinTown/GremlinTown.java

```diff
@@ -0,0 +1,652 @@
+        // (Ignore the lack of any thematic connection, it's just an example)
+        return Arrays.asList(
+                new ZoneEncounter(GREMLIN_RIDERS, 2, () -> new MonsterGroup(
+                        new AbstractMonster[] {
+                                new GremlinRiderGreen(-320.0F, -12.0F),
+                                new GremlinRiderRed(-100.0F, -30.0F),
+                                new GremlinRiderBlue(150.0F, 15.0F)
+                        })),
+                new ZoneEncounter(GREMLIN_GANG_TWO, 2, GremlinTown::getGremlinGangGroup),
+                new ZoneEncounter(NIB, 2, () -> new MonsterGroup(new GremlinNib(0f, 0f)))
+        );
+    }
+
+    private static MonsterGroup getGremlinGangGroup() {
+        ArrayList<String> gremlinPool = new ArrayList<>();
+        gremlinPool.add("ArmoredGremlin");
+        gremlinPool.add("ChubbyGremlin");
+        gremlinPool.add("GremlinAssassin");
+        gremlinPool.add("GremlinBarbarian");
+        gremlinPool.add("GremlinHealer");
+
+        AbstractMonster[] retVal = new AbstractMonster[3];
+        int index = AbstractDungeon.miscRng.random(gremlinPool.size() - 1);
+        String key = gremlinPool.get(index);
+        gremlinPool.remove(index);
+        retVal[0] = getGremlin(key, -430.0F, -25.0F);
+        index = AbstractDungeon.miscRng.random(gremlinPool.size() - 1);
+        key = gremlinPool.get(index);
+        gremlinPool.remove(index);
+        retVal[1] = getGremlin(key, -170.0F, 21.0F);
+        index = AbstractDungeon.miscRng.random(gremlinPool.size() - 1);
+        key = gremlinPool.get(index);
+        gremlinPool.remove(index);
+        retVal[2] = getGremlin(key, 115.0F, -30.0F);
+        index = AbstractDungeon.miscRng.random(gremlinPool.size() - 1);
+
+        return new MonsterGroup(retVal);
+    }
+
+    public static AbstractMonster getGremlin(String key, float xPos, float yPos) {
+        switch (key) {
+            case "GremlinAssassin":
+                return new GremlinAssassin(xPos, yPos);
+            case "GremlinBarbarian":
+                return new GremlinBarbarian(xPos, yPos);
+            case "GremlinHealer":
+                return new GremlinHealer(xPos, yPos);
+            case "ChubbyGremlin":
+                return new ChubbyGremlin(xPos, yPos);
+            case "ArmoredGremlin":
+                return new ArmoredGremlin(xPos, yPos);
+            default:
+                return new GremlinWarrior(xPos, yPos);
+        }
+    }
+
+    @Override
+    public List<ZoneEncounter> getEliteEncounters() {
+        // Override that one method so the gremlin leader encounter isn't double registered
+        return Arrays.asList(
+                GREMLIN_LEADER_ENCOUNTER,
+                new ZoneEncounter(GREMLIN_ELDER, 2, () -> new MonsterGroup(new GremlinElder(-80f, -50f))),
+                new ZoneEncounter(GREMLIN_HORDE, 2, GremlinTown::getHordeStartGroup)
```

Comment:
Fun (and funny) fight. Feels like it might put too much of an emphasis on having really good aoe and might go on too long, but it's really cool to see what you did with it. If you see ways to tone it down slightly it might be a good idea.

That said, it's hard to evaluate since this fight is doing a lot of stuff at once, and it's a cool fight, so whether or not you decide to tone it down a bit, we can say that it's fine and we'll see if we get any feedback about it.

Reply:
I've made some changes to the horde fight to try to make it not as absurdly AoE lopsided, although of course it will continue to heavily favor AoE.  I've pushed the changes but I'm probably going to be tweaking this throughout the day and maybe even after release.